Compare all branches in the tests

alesaggio opened the pull request:
similar to https://github.com/cp3-llbb/Framework/pull/262
@pieterdavid

alesaggio committed:
compare all branches

pieterdavid commented:
I'm having a look at exiting after the first event in case any branches are different (right now you get a message for every modified branch in the tree, times the number of events (10))...

pieterdavid committed:
Catch CHECK return values, and look at the whole entry

This should make sure to compare all branches that exist in both trees
before exiting

pieterdavid commented:
Running the tests on a work branch with both this PR and #268 , I get the output below for the first test, which is as expected (it reports all different branches, and only exits after the first event).
```
-------------------------------------------------------------------------------                                                                                                                           [1476/4540]
Check if trees are equals
-------------------------------------------------------------------------------
/home/users/p/d/pdavid/ttW/CMSSW_8_0_30/src/cp3_llbb/Framework/bin/testFrameworkOutput.cc:103
...............................................................................

/home/users/p/d/pdavid/ttW/CMSSW_8_0_30/src/cp3_llbb/Framework/bin/testFrameworkOutput.cc:98: FAILED:
  CHECK( same_hash )
with expansion:
  false
with message:
  [Event 0] Checking branches met_significance of type 

/home/users/p/d/pdavid/ttW/CMSSW_8_0_30/src/cp3_llbb/Framework/bin/testFrameworkOutput.cc:98: FAILED:
  CHECK( same_hash )
with expansion:
  false
with message:
  [Event 0] Checking branches met_jecdown_significance of type 

/home/users/p/d/pdavid/ttW/CMSSW_8_0_30/src/cp3_llbb/Framework/bin/testFrameworkOutput.cc:98: FAILED:
  CHECK( same_hash )
with expansion:
  false
with message:
  [Event 0] Checking branches met_jecup_significance of type 

/home/users/p/d/pdavid/ttW/CMSSW_8_0_30/src/cp3_llbb/Framework/bin/testFrameworkOutput.cc:98: FAILED:
  CHECK( same_hash )
with expansion:
  false
with message:
  [Event 0] Checking branches met_jerdown_significance of type 

/home/users/p/d/pdavid/ttW/CMSSW_8_0_30/src/cp3_llbb/Framework/bin/testFrameworkOutput.cc:98: FAILED:
  CHECK( same_hash )
with expansion:
  false
with message:
  [Event 0] Checking branches met_jerup_significance of type 

/home/users/p/d/pdavid/ttW/CMSSW_8_0_30/src/cp3_llbb/Framework/bin/testFrameworkOutput.cc:186: FAILED:
  REQUIRE( all_branches_identical )
with expansion:
  false

===============================================================================
test cases:    1 |    0 passed | 1 failed
assertions: 3512 | 3506 passed | 6 failed

Output file does not match reference file: status 6
```

OlivierBondu reviewed (approved):
Looks fine to me, thanks !
